rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,154 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
 1. Resolve peer's node DID to its corresponding DID document.
 2. Assert that one of the `NutsComm` endpoint's host matches (one of) the `dNSName` SANs in the peer's TLS client certificate.
 
-## X. Private Transactions
\ No newline at end of file
+## 4. Set reconciliation protocol
+
+### 4.1 Data requirements
+
+The protocol requires an XOR value and IBLT structure to be sent in a message. 
+These are to be calculated over different transaction ranges based on LC value.
+Both data structures have associative properties allowing for optimized data storage.
+
+#### 4.1.1 XOR
+
+The exclusive-or (XOR) value is calculated over all transactions reference within a range.
+The length of the XOR value is therefore the same as the transaction reference length (SHA256: 32 bytes).
+The XOR value is used to quickly determine if two nodes have processed the exact same set of transactions.
+
+#### 4.1.2 Invertible Bloom Lookup Table
+
+The IBLT is a data structure capable of finding differences between sets.
+The performance of this data structure is not impacted by the number of entries, but only by the size of the set difference.
+This allows two nodes to compare the entire set of transactions with a relative small data structure.
+
+An IBLT has several parameters that define its characteristics. When comparing two IBLTs, these parameters MUST be the same.
+An overview of the parameters:
+
+* **buckets**: defines the size of the IBLT, much like the size of a bloom filter. The larger the size, the bigger the set difference can be. 
+* **Hk**: each key is hashed to select the buckets the key is inserted to. The hash function is applied on the previous output to create extra hash values. This continues until k different values are found. The hash function, its seed and the value for k MUST all be the same for the entire network.
+* **Hc**: each key is hashed to a checksum which is used to validate the inserted key during the deconstruction phase.
+* **data types**: the data types for *count*, *val_sum* and *hash_sum* MUST match ((un)signed, byte-size).
+
+The following parameters are used:
+
+* **buckets**: 1024
+* **Hc**: [murmur3](https://en.wikipedia.org/wiki/MurmurHash), 8 bytes, seed: 0x00
+* **Hk**: murmur3, 4 bytes, seed: 0x01, k = 6 
+* **key sum size**: 32 bytes
+* **bucket count field**: 4 bytes
+
+See Appendix A.1 for an explanation.
+
+### 4.1.3 IBLT page size
+
+Calculating an IBLT is relatively cheap, but it grows linearly with the number of transactions. This would harm performance of larger networks.
+The immutable nature of transactions make it possible to store intermediate IBLTs and use those as an optimization.
+The LC range parameters of the protocol should be adjusted accordingly to optimize use of these intermediate IBLTs.
+Therefore, they should not be chosen freely.
+
+Any LC value used in the protocol MUST be a multiple of 512 (minus 1, because the LC is 0 based). 
```

Comment:
This reads like a TXs LC can only be 0, 521, 1042, etc. BUt I'm sure that's not what you mean?

Reply:
The LC value in the message, not on the TX